### ISSUE TITLE
[BUG] django-cms and django-modeltranslation 0.19.3+

### ISSUE THREAD

**Issue by benzkji** [django-cms/django-cms#7948]
## Description

Using these two together causes `allow_children` and `child_classes` to be ignored, and `disable_child_plugins` is always set in the structure editor. And maybe more.

## Steps to reproduce

Install django-cms 3.11.X, and django-modeltranslation 0.19.3+. Try the structure editor, plugins can be placed within others - only plugins that have a `child_classes = ["APlugin"]` work correctly. Plugins with a parent are all locked, and cnnot be moved outside the parent.

## Expected behaviour

As before, structure limits respected, no locked plugins.

## Actual behaviour

I tracked the issue to this template: https://github.com/django-cms/django-cms/blob/3.11.6/cms/templates/cms/toolbar/dragitem.html#L3

Using `{{ plugin.get_plugin_class.allow_children|pprint }}` shows `attribute error: XYPlugin has no attribute admin_site`. Not using pprint outputs the model name, i.e. always `True`. YAY. After looking around which admin could be adding some kind of getter or whatever, I found modeltranslation - 0.19.2 and everything is ok. 0.19.3 adds some typing stuff that I don't really understand, and monkeypatches Djangos ModelAdmin - as far as I get it.

Here it is, probably/maybe around the monkeypatch?!: 
https://github.com/deschler/django-modeltranslation/compare/v0.19.2...v0.19.3#diff-70244e2051407abc833c4b5691c54704040cb171edc9a46d71260332fe87ec4bR26

I don't know admin internals that well, but it seems that somewhere an admin_site is required? But where and how, no clue yet.

## Additional information (CMS/Python/Django versions)

Python 3.11, django-cms 3.11.X and django-modeltranslation 0.19.3+ to reproduce.

## Do you want to help fix this issue?

* [x] Yes, I want to help fix this issue and I will join the channel #pr-reviews on [the Discord Server](https://discord-pr-review-channel.django-cms.org) to confirm with the community that a PR is welcome.
* [ ] No, I only want to report the issue.

**Comment by fsbraun** [comment]
@benzkji Sounds like an issue with django-modeltranslation which apparently clouds the class properties that steer the plugin behavior. 

django CMS works fine with [django-parler](https://github.com/django-parler/django-parler) (which offers the additional benefit that adding languages does not create migrations).

Or pin to django-modeltranslation < 0.19.3?

**Comment by benzkji** [comment]
Yes, pinning is the workaround. @fsbraun thanks for having a look. I wonder if there are conventions about how an admin class has to be used (i.e. can it be initialized without an admin_site, that was my first thought). 

I will open an issue with modeltranslation as well.

**Comment by aacimov** [comment]
Added issue with description and images on modeltranslation issue tracker: [https://github.com/deschler/django-modeltranslation/issues/748](url)

**Comment by MacLake** [comment]
I also ran into the issue, and instead of checking out the issues on on django-modeltranslations and django-cms, I went back in my git history, and installed old version until I found the culprit. Anyway, I can confirm that it works until django-modeltranslations 0.19.2, and that 0.19.3 introduced the bug. For new apps I use django-parler, but I still have one app that had existed before I started using django-cms. I think now is a good time to replace the last uses in my project.

**Comment by benzkji** [comment]
@fsbraun it is in the depths of the template system. My guess is that modeltranslation doesn't raise a KeyError when it should, so the attributes are not fetched correctly: https://github.com/django/django/blob/main/django/template/base.py#L886

allow_children is an attribute, and needs that code to raise a KeyError, right?

**Comment by benzkji** [comment]
...TypeError, it is.

**Comment by benzkji** [comment]
There is a solution in https://github.com/deschler/django-modeltranslation/pull/755 - let's wait for a release...

**Comment by last-partizan** [comment]
I fixed this on the django-modeltranslation side, but i'd call it rather hacky solution.

In our overriden `__class_getitem__` i'm checking if class has a key with given name. So, accessing `cls["allow_children"]` fails and falls back to cls.allow_children ...

This still may fail if someone uses `django-stubs-ext` - because it does not use this hack. Maybe it's better to rewrite this somehow on the side of django-cms.

**Comment by fsbraun** [comment]
Hi @last-partizan ! Firstly, thanks so much for fixing this and secondly for reporting back. That is so helpful.

I've been thinking about this and am tempted to create an issue for Django itself. IMHO the template engine should be able to access class properties. Since the template engine checks for dictionary lookup first, it will have issues with the Python 3.10 (i believe)+ versions where classes can have dictionary lookups for typing. 

IMHO, the Django template engine should check if the dictionary lookup yields a `type` or `typing.GeneralAlias` and consider this a failed lookup (since it was a typing artefact), and then proceed to check properties.

This would solve the issue for all classes with type hints (including `django-stubs-ext`).

**Comment by benzkji** [comment]
Thank you all - I was not that deep into it, and it makes sense this way round (non hacky version - I was not aware of these "new" features).

**Comment by last-partizan** [comment]
@fsbraun by default, no class is having `__class_getitem__`, unless we inherit from `Generic` or use `class A[T]:` in newer python versions.

Maybe you're right, having that behavior in Django would solve issue, but that's probably significant change, and maybe the have their reasons why it works the way it does. But, raising awareness by creating an issue is a good idea.

But, proper way to fix this for django-cms - is rewrite the plugins in a way that does not depends on implementation details from other packages.

At least something like:

```
{% with allow_children=plugin.get_plugin_options.allow_children %}
```

Where `get_plugin_options` returns options dict/object, and not a class.

Or just moving this to context, as `"allow_children": plugin.get_plugin_class.allow_children`.

**Comment by fsbraun** [comment]
I'll raise the issue with Django and see if it gets traction. The more I think about it the more I come convinced it belongs there. It's Django that makes implicit assumptions on implementation details.

**Comment by fsbraun** [comment]
@last-partizan @benzkji Here's the Django ticket: https://code.djangoproject.com/ticket/35735

**Comment by fsbraun** [comment]
Fix for django CMS would to let `__class_getitem__` raise a `TypeError` for `CMSPluginBase´.

**Comment by fsbraun** [comment]
@last-partizan @benzkji Django 5.2 will fix the issue on Django's side: https://github.com/django/django/pull/18547 There will be no backport to earlier Django versions.

**Comment by MacLake** [comment]
This issue has been fixed in django-modeltranslation 0.19.8, so we can close this issue.

**Comment by fsbraun** [comment]
I reopen this issue, although django-modeltranslation's latest version has a workaround for django CMS. The source of the issue is[ a bug in Django's template engine](https://github.com/django/django/pull/18547), which will be fixed in Django 5.2. For Django versions below 5.2 any other 3rd party package can trigger the bug unless we ensure in django CMS that class properties are accessible. 

The fix for django CMS should contain the following bits:
* For any Django version below 5.2, add a `__class_getitem__` method to `CMSPluginBase` which raises a `TypeError`
* Regression test for Django < 5.2 that `CMSPluginBase[int]` raises a `TypeError`
